valhalla/userrequests/serializers.py

```diff
@@ -83,13 +83,14 @@ def validate(self, data):
             if data['acquire_mode'] == 'BRIGHTEST' and not data.get('acquire_radius_arcsec'):
                 raise serializers.ValidationError({'acquire_radius_arcsec': 'Acquire radius must be positive.'})
 
-        # Spectrographs must have ag_mode 'ON'
-        if configdb.is_spectrograph(data['instrument_name']):
-            if data['ag_mode'] is not 'ON':
-                raise serializers.ValidationError({'ag_mode': _('Autoguiding must be on for spectrograph observations.')})
-
         types_that_require_filter = ['expose', 'auto_focus', 'zero_pointing', 'standard', 'sky_flat']
         types_that_require_slit = ['spectrum', 'arc', 'lamp_flat']
+        types_that_require_ag_mode_on = ['spectrum', 'nres_spectrum']
```

Comment:
Can this be put in it's own validate_ag_mode() method? This validate method is big enough as it is.

Reply:
NRES_TEST should be ag_mdoe ON because it does acquisition

Reply:
NRES_EXPOSE should also be ag_mode ON because it selects the guide camera as the main instrument.

Reply:
@Fingel it can't use the built in validate_ag_mode since that only has access to the ag_mode and we need the molecule type to determine if it is valid. But I can put that logic in a helper method and call it from the main validate

Reply:
I added the nres_test and nres_expose to the list. I am not going to separate that logic for now since it is the smallest chunk of logic in the validate method (so if we want to separate things, we should separate the other chunks first)